src/Symfony/Component/PropertyInfo/Tests/Extractor/ReflectionExtractorTest.php

```diff
@@ -642,6 +642,18 @@ public function testGetWriteInfoDeprecatedEnableMagicCallExtractionInContext()
         ]);
     }
 
+    /**
+     * @requires PHP 8.1
+     */
+    public function testGetWriteInfoReadonlyProperties()
+    {
+        $writeMutatorConstructor = $this->extractor->getWriteInfo(Php81Dummy::class, 'foo', ['enable_constructor_extraction' => true]);
+        $writeMutatorWithoutConstructor = $this->extractor->getWriteInfo(Php81Dummy::class, 'foo', ['enable_constructor_extraction' => false]);
+
+        $this->assertEquals(PropertyWriteInfo::TYPE_CONSTRUCTOR, $writeMutatorConstructor->getType());
+        $this->assertEquals(PropertyWriteInfo::TYPE_NONE, $writeMutatorWithoutConstructor->getType());
```

Comment:
should use asserSame (please fetch before pushing again as I force-pushed on your fork)